daemon/daemon.go

```diff
@@ -309,7 +313,7 @@ func (d *OrderedDaemon) stopWorkers() {
 			if d.logger != nil {
 				d.logger.Debugf("Stopping Background Worker: %s ...", name)
 			}
-			close(worker.shutdownSignal)
+			worker.ctxCancel()
```

Comment:
Since subsequent calls to `CancelFunc()` do nothing, I think it would be fine to also call it in the `if !worker.running.IsSet() ...` clause to simply ensure that the worker really gets a termination signal.